IMOS/finaliseData.m

```diff
@@ -161,47 +161,28 @@
       end
   end
   
-  if isempty(sam.time_deployment_start)
-      sam.time_deployment_start         = [];
-      sam.time_deployment_start_origin  = [];
-  end
-  if isempty(sam.time_deployment_end)
-      sam.time_deployment_end           = [];
-      sam.time_deployment_end_origin    = [];
-  end
-  
-  % set the time coverage period from the data
-  switch mode
-      case 'profile'
-          iTime = getVar(sam.variables, 'TIME');
-          if iTime ~= 0
-              if isempty(sam.time_coverage_start),
-                  sam.time_coverage_start = sam.variables{iTime}.data(1);
-              end
-              if isempty(sam.time_coverage_end),
-                  sam.time_coverage_end   = sam.variables{iTime}.data(end);
-              end
-          else
-              if isempty(sam.time_coverage_start), sam.time_coverage_start = []; end
-              if isempty(sam.time_coverage_end),   sam.time_coverage_end   = []; end
-          end
-          
-      case 'timeSeries'
-          iTime = getVar(sam.dimensions, 'TIME');
-          if iTime ~= 0
-              if isempty(sam.time_coverage_start),
-                  sam.time_coverage_start = sam.dimensions{iTime}.data(1);
-              end
-              if isempty(sam.time_coverage_end),
-                  sam.time_coverage_end   = sam.dimensions{iTime}.data(end);
-              end
-          else
-              if isempty(sam.time_coverage_start), sam.time_coverage_start = []; end
-              if isempty(sam.time_coverage_end),   sam.time_coverage_end   = []; end
-          end
-  
+  % set the time coverage/deployment period from the data
+  iTime = getVar(sam.variables, 'TIME');
```

Comment:
I disagree with this refactoring. In profile mode TIME is in sam.variables while in timeseries mode TIME is in sam.dimensions. 
If you really want to refactor you could declare a string modeStr that you set either to 'variables' or 'dimensions' depending on the mode:
```Matlab
switch mode
  case 'profile'
    modeStr = 'variables';
  case 'timeSeries'
    modeStr = 'dimensions';
end

iTime = getVar(sam.(modeStr), 'TIME')
```
 ... etc

Reply:
I forgot that - good catch. It would be nice to fix this mess-up - TIME is always a variable so it should only be store in the variable fieldname. I shall raise a small refactoring issue for that.

I implemented the fix with ```if``` so it is shorter and more explicit. Tests are also added - completely forgot to run/include them.